File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        freq = {}
        ans = []
        # character with their freq in freq
        for ch in s:
            freq[ch] = freq.get(ch,0)+1
        for _ in range(len(freq)):
            maxFreq = 0
            ke=''
            for ch in freq:
                if(freq.get(ch)>maxFreq):
                    maxFreq=freq.get(ch)
                    ke = ch
            # now add in the ans's list
            ans.append(ke*maxFreq)
            # mark ke is used
            freq[ke]=0
        # return ans
        return ''.join(ans) 

        
        

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {
        HashMap<Character,Integer> map = new HashMap<>();
        StringBuilder sb = new StringBuilder("");
        for(char ch:s.toCharArray()){
            map.put(ch,map.getOrDefault(ch,0)+1);
        }
        for(int i=0;i<map.size();i++){
            int max=0;
            char k=' ';
            int val=0;
            for(char c:map.keySet()){
                if(max<map.get(c)){
                    max=map.get(c);
                    val=max;
                    k= c;
                }
            } // inner loop ends
            // add into sb
            for(int tm =0;tm<val;tm++){
                sb.append(k);
            }
            // now the key(k) with value '0' in HashMap so other character can come
            map.put(k,0);
        }
        return sb.toString();
    }
}